src/mgmt/cli_mgmt.c

```diff
@@ -625,14 +641,52 @@ dc_mgmt_net_cfg(const char *name, crt_init_options_t *crt_info)
 	D_STRNDUP(crt_info->cio_provider, info->provider, DAOS_SYS_INFO_STRING_MAX);
 	if (NULL == crt_info->cio_provider)
 		D_GOTO(cleanup, rc = -DER_NOMEM);
-	D_STRNDUP(crt_info->cio_interface, info->interface, DAOS_SYS_INFO_STRING_MAX);
-	if (NULL == crt_info->cio_interface)
-		D_GOTO(cleanup, rc = -DER_NOMEM);
-	D_STRNDUP(crt_info->cio_domain, info->domain, DAOS_SYS_INFO_STRING_MAX);
-	if (NULL == crt_info->cio_domain)
-		D_GOTO(cleanup, rc = -DER_NOMEM);
 
-	D_INFO("Network interface: %s, Domain: %s\n", info->interface, info->domain);
+	d_getenv_bool("D_DYNAMIC_CTX", &d_dynamic_ctx_g);
+	if (d_dynamic_ctx_g) {
+		int i, cnt = 0;
+
+		for (i = 0; i < resp->n_numa_fabric_interfaces; i++)
+			cnt += resp_g->numa_fabric_interfaces[i]->n_ifaces + 1;
+
+		D_ALLOC(crt_info->cio_interface, DAOS_SYS_INFO_STRING_MAX * cnt);
+		if (crt_info->cio_interface == NULL)
+			D_GOTO(cleanup, rc = -DER_NOMEM);
+		D_ALLOC(crt_info->cio_domain, DAOS_SYS_INFO_STRING_MAX * cnt);
+		if (crt_info->cio_domain == NULL)
+			D_GOTO(cleanup, rc = -DER_NOMEM);
+
+		for (i = 0; i < resp->n_numa_fabric_interfaces; i++) {
+			Mgmt__FabricInterfaces *numa_ifaces = resp_g->numa_fabric_interfaces[i];
+			int                     j;
+
+			for (j = 0; j < numa_ifaces->n_ifaces; j++) {
+				if (i != 0 || j != 0) {
+					strcat(crt_info->cio_interface, ",");
+					strcat(crt_info->cio_domain, ",");
+				}
+				strcat(crt_info->cio_interface, numa_ifaces->ifaces[j]->interface);
+				strcat(crt_info->cio_domain, numa_ifaces->ifaces[j]->domain);
```

Comment:
Just occurred to me, we probably should use `strncat` here, as we know the length of the destination buffer. Coverity will probably ding us for `strcat`.

Reply:
no need because we should have already verified the in strings since they are coming from the resp. so it can be marked as false positive in coverity.

Reply:
since im already repushing ill just incorporate that